server/controllers/index.js

```diff
@@ -0,0 +1,33 @@
+const AWS = require('aws-sdk');
+const { uploadAudioFile, uploadImageFile, s3 } = require('../s3');
+const models = require('../models/index');
+
+const handleUpload = (req, res) => {
+  const data = req.body;
+  const { tags } = req.body;
+  const audioFileData = req.files['audioFile'][0].buffer;
+  const imageFileData = req.files['imageFile'][0].buffer;
+  let audioUrl;
+  let imageUrl;
+  uploadAudioFile(audioFileData)
+    .then(audioFileUrl => {
+      data.path_to_song = audioFileUrl;
+      audioUrl = audioFileUrl;
+      return uploadImageFile(imageFileData);
+    })
+    .then(imageFileUrl => {
+      data.path_to_artwork = imageFileUrl;
+      imageUrl = imageFileUrl;
+      models.addSong(data);
+      models.addTags(tags, req.body.title);
+      res.json({ audioUrl, imageUrl });
+    })
+    .catch(error => {
+      console.error(error);
+      res.status(500).json({ error: 'Failed to upload audio file' });
+    });
+};
+
+module.exports = {
+  handleUpload
+};
```

Comment:
I wonder if naming this file index.js is a little confusing. I like having the base file inside the server folder, the db folder, and the client src folder be index.js, but it feels like now we're just naming folders and naming every module index.js. This will probably get very confusing. Why not name this handleUpload.js?

Reply:
I was going to add all controller functions to this file, but we can split our controllers into multiple files and just import all of the files into our server/index.js file!